src/js/background.js

```diff
@@ -46,39 +46,44 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
             sendResponse({
               response: 'paratii.success.ok'
             })
-
-            // let uri = request.payload.uri
-            // console.log(uri)
-            // node.files.cat(uri, (err, stream) => {
-            //   if (err) {
-            //     console.log('IPFS cat error')
-            //   }
-            //   stream.on('data', (data) => {
-            //     console.log('Received data packet')
-            //     console.log(data)
-            //     chrome.tabs.query({
-            //       active: true
-            //     }, (tabs) => {
-            //       tabs.forEach((tab) => {
-            //         chrome.tabs.sendMessage(tab, {
-            //           type: 'PARATII_DATA',
-            //           requestID: uri,
-            //           data: data
-            //         }, (response) => {
-            //           console.log('Received response to data packet...')
-            //           console.log(response)
-            //           // onDataResponseHandler(response)
-            //         })
-            //       })
-            //     })
-            //   })
-            // })
           })
         })
       }
     } else if (request.payload.action === 'paratii.stop') {
       console.log('Stopping IPFS')
       node.stop()
+    } else if (request.payload.action === 'paratii.upload') {
+      console.log('Got msg with action upload')
+      console.log(request.payload)
+      let buffer = request.payload.buffer
+      console.log('buffer should be', buffer)
+      let file = {
+        path: 'video.mp4',
+        content: buffer
+      }
+      console.log('file should be ', file)
+      node.files.add(file, (err, result) => { // Upload buffer to IPFS
```

Comment:
**Don't change this now. this is just an FYI**
This is probably not suitable for large (+50MB) files, here how we do it in paratii-player uploader https://github.com/Paratii-Video/paratii-player/blob/master/imports/ui/pages/upload/ipfs-uploader.js#L62

**note** Pull streams are kind of weird in the beginning 😄